libkbfs/cr_chains.go

```diff
@@ -86,10 +86,9 @@ func (cc *crChain) collapse(createdOriginals map[BlockPointer]bool,
 			// The last op will have its refs listed twice in the
 			// collapsed op, but that's harmless.
 			syncRefs = append(syncRefs, op.Refs()...)
-			// The last op will have its unrefs listed twice, once in
-			// the collapsed op and once in the resolutionOp, but
-			// that's harmless.
-			toUnrefs = append(toUnrefs, op.Unrefs()...)
+			// The last op will have its unrefs listed twice in the
+			// collapsed op, but that's harmless.
+			syncUnrefs = append(syncUnrefs, op.Unrefs()...)
```

Comment:
If we include `syncOp.Unrefs()` in `toUnrefs`, then those blocks will end up getting checked against the journal's unflushed blocks.  If they haven't been flushed yet, they will be deleted from the journal and not unref'd.  However, that's not what we want to happen for SyncAll because it would be causing a block from some previous journal update to not get flushed.  For a normal squash/CR this is ok, because any blocks that show up as unrefs that can be safely deleted will show up as keys in the `createdOriginals` in the chain, and still be checked against the unflushed journal.

It's entirely possible I'll have to revisit the rest of the data in `toUnrefs` once I add directory operations to the batch.  But I didn't want to mess with it too much right now.